src/transaction/boot_cl.c

```diff
@@ -6720,6 +6723,255 @@ boot_define_view_synonym (void)
   return NO_ERROR;
 }
 
+#define SELECT_OBJECTS_QUERY  "SELECT \
+      /* CLASS/VCLASS */ \
+       [c].[owner_name] AS [owner], \
+       [c].[class_name] AS [object_name], \
+       CASE [c].[class_type] \
+         WHEN 'CLASS' THEN 'table' \
+         WHEN 'VCLASS' THEN 'view' \
+         ELSE 'unknown' \
+       END AS [object_type], \
+       CASE \
+         WHEN [c].[is_system_class] = 'YES' THEN 'Y' \
+         WHEN [c].[partitioned] = 'YES' THEN 'Y' \
+         ELSE 'N' \
+       END AS [generated] \
+    FROM \
+       [db_class] AS [c] \
+   UNION ALL  \
+    /* INDEX */ \
+    SELECT \
+       [i].[owner_name] AS [owner], \
+       [i].[index_name] AS [object_name], \
+       'index'          AS [object_type], \
+       CASE \
+         WHEN [i].is_primary_key = 'YES' THEN 'Y' \
+         WHEN [i].is_foreign_key = 'YES' THEN 'Y' \
+         ELSE 'N' \
+       END AS [generated] \
+    FROM \
+       [db_index] AS [i] \
+   UNION ALL  \
+    /* TRIGGER */ \
+    SELECT \
+       [t].[target_owner_name] AS [owner], \
+       [t].[trigger_name]      AS [object_name], \
+       'trigger'               AS [object_type], \
+       'N'                     AS [generated] \
+    FROM \
+       [db_trig] AS [t] \
+   UNION ALL  \
+    /* SERIAL */ \
+    SELECT \
+       'PUBLIC'   AS [owner], \
+       [s].[name] AS [object_name], \
+       'serial'   AS [object_type], \
+       'N'        AS [generated] \
```

Comment:
db_serial can include generated attributes for "AUTO_INCREMENT".

Reply:
IMO, along with partition tables, The auto_increment information also should be added in system catalog too.